src/common/Pager/index.js

```diff
@@ -5,29 +5,46 @@ import Next from "./images/Next.svg"
 import DisabledNext from "./images/DisabledNext.svg"
 import { PagerButton, PagerNext, PagerPrev, PagerText, PagerInfo, PagerWrapper, Bold } from "./styled"
 import { useDispatch, useSelector } from "react-redux";
-import { selectCurrentMoviesPage, incrementMoviesPage, decrementMoviesPage, selectAllMoviesPages, setMoviesPageToFirst, setMoviesPageToLast } from "../../features/movies/moviesSlice";
+import { selectAllMoviesPages, setMoviesPage } from "../../features/movies/moviesSlice";
 import { theme } from "../../theme";
-import {useLocation} from "react-router-dom";
-import { selectCurrentPeoplePage, selectPeopleAllPages, incrementPeoplePage, decrementPeoplePage, setPeoplePageToFirst, setPeoplePageToLast } from "../../features/people/peopleSlice"
+import { selectPeopleAllPages, setPeoplePage } from "../../features/people/peopleSlice";
+import { useLocation } from "react-router-dom";
+import { useQueryParameter, useReplaceQueryParameter } from "../../features/search/queryParameters";
+import pageParam from "../../features/pageQueryParamName";
+
 
 export const Pager = () => {
     const mobileWidth = theme.breakpoint.mobileMax;
     const location = useLocation();
+    const dispatch = useDispatch();
     const atMovies = location.pathname.includes("movies") ? true : false;
-    const currentMoviesPage = useSelector(selectCurrentMoviesPage);
+    const currentPage = +useQueryParameter(pageParam) || 1;
+    dispatch(atMovies ? setMoviesPage(currentPage) : setPeoplePage(currentPage));
```

Comment:
tak raczej nie można sobie dispatchować akcji wewnątrz koponentu, trzeba to zrobić w useEffect jako efekt uboczny zmiany parametru currentPage czyli: 
useEffect( () => {
...
}, [currentPage]);